src/object_store/src/object/opendal_engine/fs.rs

```diff
@@ -17,14 +17,18 @@ use opendal::services::Fs;
 use opendal::Operator;
 
 use super::{EngineType, OpendalObjectStore};
+use crate::object::opendal_engine::ATOMIC_WRITE_DIR;
 use crate::object::ObjectResult;
+
 impl OpendalObjectStore {
     /// create opendal fs engine.
     pub fn new_fs_engine(root: String) -> ObjectResult<Self> {
         // Create fs backend builder.
         let mut builder = Fs::default();
 
         builder.root(&root);
+        let atomic_write_dir = format!("{}{}", root, ATOMIC_WRITE_DIR);
```

Comment:
Do we guarantee that `root` always ends with the separator (`/`)?

Reply:
Yes, root always ends with `/`, and let's just set atomic_write_dir `data_directory/atomic_write_dir`, thus will not  affect the data under `root`. (Actually the `root` is always `/` now).

Reply:
I have two questions:
1. Will the temporary files in `data_directory/atomic_write_dir` be deleted if write fails? We may need to check OpenDAL implementation to confirm.
2. Will files in `data_directory/atomic_write_dir` eligible for full gc? cc @zwang28. If these files have `.data` suffix, then I think the answer is yes (See [here](https://github.com/risingwavelabs/risingwave/blob/2e614b9f21494903fd03ec454c13d85418870022/src/storage/src/hummock/sstable_store.rs#L666)). If that is the case, we should blacklist `atomic_write_dir` for full gc.

Reply:
> 1. no

oh.... Then maybe it makes sense to let full gc take care of the files in `atomic_write_dir`. No changes are needed.

Reply:
> I have two questions:
> 
> 1. Will the temporary files in `data_directory/atomic_write_dir` be deleted if write fails? We may need to check OpenDAL implementation to confirm.
> 2. Will files in `data_directory/atomic_write_dir` eligible for full gc? cc @zwang28. If these files have `.data` suffix, then I think the answer is yes (See [here](https://github.com/risingwavelabs/risingwave/blob/2e614b9f21494903fd03ec454c13d85418870022/src/storage/src/hummock/sstable_store.rs#L666)). If that is the case, we should blacklist `atomic_write_dir` for full gc.

I'm also assisting in this investigation.

Reply:
> > I have two questions:
> > 
> > 1. Will the temporary files in `data_directory/atomic_write_dir` be deleted if write fails? We may need to check OpenDAL implementation to confirm.
> > 2. Will files in `data_directory/atomic_write_dir` eligible for full gc? cc @zwang28. If these files have `.data` suffix, then I think the answer is yes (See [here](https://github.com/risingwavelabs/risingwave/blob/2e614b9f21494903fd03ec454c13d85418870022/src/storage/src/hummock/sstable_store.rs#L666)). If that is the case, we should blacklist `atomic_write_dir` for full gc.
> 
> I'm also assisting in this investigation.

What do you mean?